packages/cli/src/lib/git.ts

```diff
@@ -155,8 +180,34 @@ export async function findBaseCommit(headRef = 'HEAD', cwd?: string): Promise<st
  * @param {string} [cwd] - The working directory to run the command in.
  * @returns {Promise<string>} A promise containing the diff.
  */
-export function getWorkingDiff(cwd?: string): Promise<string> {
-  return execute('git', ['diff', 'HEAD'], { cwd });
+export async function getWorkingDiff(cwd?: string): Promise<string> {
+  const untrackedFiles = await getUntrackedFiles(cwd);
+  const untrackedDiffs: string[] = [];
+
+  // Do this serially to avoid spawning too many processes.
+  // TODO: Perhaps this could be done in a queue.
```

Comment:
From our experience building the file search index, nothing fancier than a loop like this really seems to be needed.